packages/core/graph/src/AdjacencyList.js

```diff
@@ -254,6 +260,10 @@ export default class AdjacencyList<TEdgeType: number = 1> {
     // The edge is already in the graph; do nothing.
     if (edge !== null) return false;
 
+    if (!this.#edgeTypes.includes(type)) {
```

Comment:
[Guess this is what you mean for the](https://github.com/parcel-bundler/parcel/blob/a171677e90829b4733397160ad177890cb6f2cee/packages/core/core/src/BundleGraph.js#L42 ) `BundleGraph` ?

I'll have a look into doing it this way. I agree it's more elegant.

Reply:
> We just weren't sure there was a point of truth for all edge types for all the graphs? I do think the costs of serializing for this Array would be insignificant though.

Yeah good point since the list of edge types for any given graph is pretty small though I do think that determining what all the edge types are for a graph inside AdjacencyList is a little weird. Makes more sense to me to pass in each graph's possible edge types into AdjacencyList 🤷 but I think either solution is fine

> [Guess this is what you mean for the](https://github.com/parcel-bundler/parcel/blob/a171677e90829b4733397160ad177890cb6f2cee/packages/core/core/src/BundleGraph.js#L42) BundleGraph ?

Yep! These are the [ones for RequestGraph](https://github.com/parcel-bundler/parcel/blob/a171677e90829b4733397160ad177890cb6f2cee/packages/core/core/src/RequestTracker.js#L54-L61), and `AssetGraph` doesn't use edge types